github/commands/open_on_remote.py

```diff
@@ -13,17 +14,67 @@ class GsOpenFileOnRemoteCommand(TextCommand, GitCommand, git_mixins.GithubRemote
     """
     Open a new browser window to the web-version of the currently opened
     (or specified) file. If `preselect` is `True`, include the selected
-    lines in the request.
+    lines in the request. If the active tracked remote is the same as the
+    integrated remote, open browser directly, if not, display a list to remotes
+    to choose from.
 
     At present, this only supports github.com and GitHub enterprise.
     """
 
-    def run(self, edit, preselect=False, fpath=None):
-        fpath = fpath or self.get_rel_path()
+    def run(self, edit, remote=None, preselect=False, fpath=None):
+        self.fpath = fpath or self.get_rel_path()
+        self.preselect = preselect
+
+        self.remotes = self.get_remotes()
```

Comment:
This (and later commands) should run on the non-UI thread, so as to avoid blocking user interaction.  Look for anywhere else we're using `set_timeout_async` and `run_async` pattern.